src/util-decode-mime.c

```diff
@@ -1311,89 +1330,83 @@ static int ProcessBase64BodyLine(const uint8_t *buf, uint32_t len,
                 len, MAX_ENC_LINE_LEN);
     }
 
+    if (state->bvr_len + len < B64_BLOCK) {
+        memcpy(state->bvremain + state->bvr_len, buf, len);
+        state->bvr_len += len;
+        len = 0;
+    }
+
     /* First process remaining from previous line */
     if (state->bvr_len > 0) {
-
-        SCLogDebug("Base64 line remainder found: %u", state->bvr_len);
-
         /* Process remainder and return number of bytes pulled from current buffer */
         rem1 = ProcessBase64Remainder(buf, (uint8_t)len, state, 0);
+        int32_t remainder_b64 = len - rem1;
+        if (remainder_b64 < B64_BLOCK) {
```

Comment:
I see that `remainder_b64` can be 0, which leads to a useless memcpy call. But what does it mean if it is 0?